demos/CMakeLists.txt

```diff
@@ -63,7 +50,7 @@ if (WIN32)
         set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /WX") #treating warnings as errors
     endif ()
 
-    if (${CMAKE_CXX_COMPILER_ID} STREQUAL MSVC)
+    if (CMAKE_CXX_COMPILER_ID MATCHES MSVC)
```

Comment:
Why use `MATCHES` here? The ID should be exactly `MSVC`.

Reply:
@ilya-lavrenov , could you please clarify why you used `MATCHES` here? According to the https://cmake.org/cmake/help/v3.0/command/if.html link the MATCHES keyword is used to check regexp validity. But `STREQUAL` is used to verify lexicographical equality. Do you use it here intentionally?

Reply:
I think I know why, actually. `MSVC` is also the name of a CMake variable, so `CMAKE_CXX_COMPILER_ID STREQUAL MSVC` compares the values of the `CMAKE_CXX_COMPILER_ID` and `MSVC` variables, rather than comparing the value of `CMAKE_CXX_COMPILER_ID` to the string "MSVC".

There are better ways to fix this, though:

1. Enable policy `CMP0054` and use `if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")`, or...
2. Just use `if(MSVC)` instead.